spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * @param <T> MQTT client type
+ * @param <C> MQTT connection options type (v5 or v3)
+ *
+ * @author Artem Vozhdayenko
+ *
+ * @since 6.0
+ */
+public abstract class AbstractMqttClientManager<T, C> implements ClientManager<T, C>, ApplicationEventPublisherAware {
+
+	protected final Log logger = LogFactory.getLog(this.getClass()); // NOSONAR
+
+	private static final int DEFAULT_MANAGER_PHASE = 0;
+
+	private final Set<ConnectCallback> connectCallbacks = Collections.synchronizedSet(new HashSet<>());
+
+	private final String clientId;
+
+	private int phase = DEFAULT_MANAGER_PHASE;
+
+	private boolean manualAcks;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private String url;
+
+	private volatile T client;
+
+	private String beanName;
+
+	AbstractMqttClientManager(String clientId) {
+		Assert.notNull(clientId, "'clientId' is required");
+		this.clientId = clientId;
+	}
+
+	protected void setManualAcks(boolean manualAcks) {
+		this.manualAcks = manualAcks;
+	}
+
+	protected String getUrl() {
+		return this.url;
+	}
+
+	protected void setUrl(String url) {
+		this.url = url;
+	}
+
+	protected String getClientId() {
+		return this.clientId;
+	}
+
+	protected ApplicationEventPublisher getApplicationEventPublisher() {
+		return this.applicationEventPublisher;
+	}
+
+	protected synchronized void setClient(T client) {
+		this.client = client;
+	}
+
+	protected Set<ConnectCallback> getCallbacks() {
+		return this.connectCallbacks;
+	}
+
+	@Override
+	public boolean isManualAcks() {
+		return this.manualAcks;
+	}
+
+	@Override
+	public T getClient() {
+		return this.client;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		Assert.notNull(applicationEventPublisher, "'applicationEventPublisher' cannot be null");
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	@Override
+	public void setBeanName(String name) {
+		this.beanName = name;
+	}
+
+	@Override
+	public String getBeanName() {
+		return this.beanName;
+	}
+
+	/**
+	 * The phase of component autostart in {@link SmartLifecycle}.
+	 * If the custom one is required, note that for the correct behavior it should be less than phase of
+	 * {@link AbstractMqttMessageDrivenChannelAdapter} implementations.
+	 * The default phase is {@link #DEFAULT_MANAGER_PHASE}.
+	 * @return {@link SmartLifecycle} autostart phase
+	 * @see #setPhase
+	 */
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	@Override
+	public void addCallback(ConnectCallback connectCallback) {
+		this.connectCallbacks.add(connectCallback);
+	}
+
+	@Override
+	public boolean removeCallback(ConnectCallback connectCallback) {
+		return this.connectCallbacks.remove(connectCallback);
+	}
+
+	public synchronized boolean isRunning() {
+		return this.client != null;
+	}
+
+	/**
+	 * Sets the phase of component autostart in {@link SmartLifecycle}.
```

Comment:
The method Javadoc must be *imperative*, not *infinitive*: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting